packages/cozy-harvest-lib/src/helpers/accounts.js

```diff
@@ -94,6 +96,55 @@ export const mergeAuth = (account, authData) => ({
   auth: merge(account.auth, authData)
 })
 
+/**
+ * Gets the vault cipher relationship to an account
+ * @param  {object} account   io.cozy.accounts document
+ * @return {string}           cipher uuid
+ */
+export const getVaultCipherId = account =>
+  account.relationships &&
+  account.relationships.vaultCipher &&
+  account.relationships.vaultCipher._id
+
+/**
+ * Gets the linked cipher in vault
+ * If none is linked, search on that match
+ * @param account - io.cozy.accounts document
+ * @param {WebVaultClient} vaultClient
+ * @param {object} search criterias
+ * @param {string} search.username
+ * @param {string} search.password
+ * @param {string} search.uri
+ * @return {CipherView|null}
+ */
+export const getVaultCipher = async (
+  account,
+  vaultClient,
+  { username, password, uri } = {}
+) => {
+  const id = getVaultCipherId(account)
+  if (id) {
+    // there is an explicit existing link
+    const cipher = await vaultClient.get(id)
+    if (cipher) return cipher
```

Comment:
If teh cipher can't be fond, shoud we re-attach the account to the new cipher?

Reply:
I think so